flow/workflows/cdc_flow.go

```diff
@@ -616,6 +619,7 @@ func CDCFlowWorkflow(
 				return nil, err
 			}
 			if setupFlowError != nil {
+				syncStatusToCatalog(ctx, workflow.GetLogger(ctx), protos.FlowStatus_STATUS_FAILED)
```

Comment:
should we update the status on line 617 for TerminateSignal? similarly at 720

Reply:
If drop flow source or destination is hung, then catalog state cleanup can take time